src/components/include/crt_lock.h

```diff
@@ -0,0 +1,59 @@
+#ifndef CRT_LOCK_H
+#define CRT_LOCK_H
+
+/***
+ * Simple blocking lock. Uses blockpoints to enable the blocking and
+ * waking of contending threads. This has little to no intelligence,
+ * for example, not expressing dependencies for PI.
+ */
+
+#include <cos_component.h>
+#include <crt_blkpt.h>
+
+struct crt_lock {
+	unsigned long owner;
+	struct crt_blkpt blkpt;
+};
+
+static inline int
+crt_lock_init(struct crt_lock *l)
+{
+	l->owner = 0;
+
+	return crt_blkpt_init(&l->blkpt);
+}
+
+static inline int
+crt_lock_teardown(struct crt_lock *l)
+{
+	assert(l->owner == 0);
+
+	return crt_blkpt_teardown(&l->blkpt);
+}
+
+static inline void
+crt_lock_take(struct crt_lock *l)
```

Comment:
Maybe we should add some kind of retry limit here... In case there is great contention.